storage_controller/src/persistence.rs

```diff
@@ -675,6 +686,130 @@ impl Persistence {
         )
         .await
     }
+
+    /// Stores all the latest metadata health updates durably. Updates existing entry on conflict.
+    ///
+    /// **Correctness:** `metadata_health_updates` should all belong the tenant shards managed by the storage controller.
+    #[allow(dead_code)]
+    pub(crate) async fn update_metadata_health_records(
+        &self,
+        metadata_health_records: Vec<MetadataHealthPersistence>,
+    ) -> DatabaseResult<()> {
+        use crate::schema::metadata_health::dsl::*;
+
+        self.with_measured_conn(
+            DatabaseOperation::UpdateMetadataHealth,
+            move |conn| -> DatabaseResult<_> {
+                for update in &metadata_health_records {
+                    diesel::insert_into(metadata_health)
+                        .values(update)
```

Comment:
I think you can insert/update multiple rows in one transaction. `[values](https://docs.diesel.rs/1.4.x/diesel/query_builder/struct.IncompleteInsertStatement.html#method.values)` takes a plural arg.

Reply:
They will still be in the same transaction but different SQL statement. Addressed in 8cf537dc4442d39b41e8fb03504e22946b6d5de1, but need two insert statements (one for all the healthy records, one for all the unhealthy records) to fight with diesel `update .. set` syntax.

After we switch to batch, we no longer can directly access the individual `health` and `last_scrubbed_at` fields.